faq.md

```diff
@@ -499,6 +499,90 @@ If you are using the Flutter plugin for IntelliJ, you can launch the app in
+
+* Reactive programming: The widget and element trees are sometimes described as reactive, because new inputs 
+provided in a widget's constructor are immediately propagated as changes to lower-level widgets by
+the widget's build method, and changes made in the lower widgets (e.g. in response to user input) propagate
+back up the tree via event handlers. Aspects of both functional-reactive and imperative-reactive are
+present in the framework. 
+
+* Declarative programming: The build functions of widgets are typically a single expression with multiple levels
+of nested constructors, written using a strictly declarative subset of Dart (such nested expressions could be mechanically
+transformed to or from any suitably expressive markup language). This can also be combined with
+the imperative style to build UIs that would be harder to describe in a pure-declarative approach.
+
+* Generic programming: Types can be used to help catch programming errors early. The Flutter framework
+uses generic programming to help in this regard. For example, the
+[State](https://docs.flutter.io/flutter/widgets/State-class.html) class is parameterized in terms
+of the type of its associated widget, so that the Dart analyzer can catch mismatches of states and widgets.
+Similarly, the [GlobalKey](https://docs.flutter.io/flutter/widgets/GlobalKey-class.html) class takes a type
+parameter so that it can access a remote widget's state in a type-safe manner (with runtime checking),
+the [Route](https://docs.flutter.io/flutter/widgets/Route-class.html) interface is parameterized with the
+type that it is expected to use when [popped](https://docs.flutter.io/flutter/widgets/Navigator/pop.html),
+and collections such as [List](https://docs.flutter.io/flutter/dart-core/List-class.html)s,
+[Map](https://docs.flutter.io/flutter/dart-core/Map-class.html)s, and
+[Set](https://docs.flutter.io/flutter/dart-core/Set-class.html)s are all parameterized so that mismatched
+elements can be caught early either by the analyzer or at runtime during debugging.
+
+* Concurrent programming: Flutter makes heavy use of
+[Future](https://docs.flutter.io/flutter/dart-async/Future-class.html)s and other asynchronous APIs.
+For example, the animation system reports when an animation is finished by completing a future. The
+image loading system similarly uses futures to report when a load is complete.
+
+* Constraint programming: The layout system in Flutter uses a weak form of constraint programming to
+determine the geometry of a scene. Constraints (e.g. for cartesian boxes, a minimum and maximum width
+and a minimum and maximum height) are passed from parent to child, and the child selects a resulting
+geometry (e.g. for cartesian boxes, a size, specifically a width and a height) that fulfills those
+constraints. By using this technique, Flutter can usually lay out a whole scene with a single pass.
```

Comment:
Good benefit. I might even add "in Order N time where N is the number of widgets".

Reply:
That's probably too strong a claim to make in the FAQ (it depends on much more, e.g. on whether you use the intrinsic dimension APIs, or whether you have off-stage widgets, etc).